modules/Kconfig.sof

```diff
@@ -9,3 +9,9 @@ config SOF
 	depends on ZEPHYR_SOF_MODULE
 	help
 	  Build Sound Open Firmware (SOF) support.
+
+
+if SOF
+config SYSTEM_CLOCK_MP_BROADCAST
+	default n
```

Comment:
This is essentially the way @lyakh had it in the original PR, and @teburd and I nak'd it because it means that "Zephyr" is testing a different timer driver configuration than "SOF" uses, which is counterproductive since the only reason to be unit testing this driver in Zephyr is to validate it for use in SOF.   (Summon @marc-hb for anecdotes about incomplete unit testing).

I think what we really want here is to identify the specific Zephyr tests that are making assumptions about timer interrupt delivery and set this for them only.  And even then we should probably work at fixing the tests.

Reply:
it was nak'd because of "#ifdef CONFIG_SOF" in driver code. 
This patch does basically the same in a different way, masking the SOF part out of the drivers.

Fishing for all possible tests, and they are many and adding this platform and SMP specific Kconfig and all of those is just wrong  IMO and does not scale, we will be playing whackamole for some time. 

Either this is needed and is a valid setup or it is not, if it is is not, then we should be changing our tests instead of just making the system behave differently for the tests.

Either way, consider this a hotfix/workaround for now, because without it 10s of tests are broken, we do need a better solution.

Reply:
I'm fine with getting this in as a regression fix, but for sure the "better solution" is to fix the tests not to be making implicit assumptions about timer interrupt delivery (or at least identify the ones doing it for some kind of justifiable reason -- there's weird trickery in some of the more white-boxy MP tests).

Again, the end goal should be that unit tests for SOF-targetted hardware like audio DSPs should be run using the SOF configuration as much as practical.

Reply:
> Again, the end goal should be that unit tests for SOF-targetted hardware like audio DSPs should be run using the SOF configuration as much as practical.

I fully agree and thats why I do not want to tinker with tests and have them behave differently using some Kconfig, if the tests need to be changed, it needs to happen at a lower level in the 1cpu variant @peter-mitsis  is talking about. This is just reverting back to old behaviour (for !SOF) to deal with a large number of tests failing.

Reply:
@peter-mitsis  you said you have a fix?

Reply:
I am working on a fix to ztest. Assuming that I am correct in thinking that CPU0 is executing a busy-wait loop (see ztest.c::cpu_hold()) with interrupts locked, I am working on a method to prevent ztest.c::cpu_hold() from executing that busy-wait loop on CPU0.

It is a little kludgy at the moment because I am assuming that I should not use pinned threads or CPU masks.